biz.aQute.bndlib/src/aQute/bnd/http/HttpClient.java

```diff
@@ -300,42 +300,61 @@ TaggedData doCached0(final HttpRequest<?> request) throws Exception, IOException
 	}
 
 	public TaggedData send0(final HttpRequest<?> request) throws Exception {
+		int retries = 0;
+		if ("GET".equalsIgnoreCase(request.verb))
+			retries = request.retries;
+
+		int trial = 0;
+		while (true) {
+			final ProxySetup proxy = getProxySetup(request.url);
+			final URLConnection con = getProxiedAndConfiguredConnection(request.url, proxy);
+			final HttpURLConnection hcon = (HttpURLConnection) (con instanceof HttpURLConnection ? con : null);
+
+			if (request.ifNoneMatch != null) {
+				request.headers.put("If-None-Match", entitytag(request.ifNoneMatch));
+			}
 
-		final ProxySetup proxy = getProxySetup(request.url);
-		final URLConnection con = getProxiedAndConfiguredConnection(request.url, proxy);
-		final HttpURLConnection hcon = (HttpURLConnection) (con instanceof HttpURLConnection ? con : null);
-
-		if (request.ifNoneMatch != null) {
-			request.headers.put("If-None-Match", entitytag(request.ifNoneMatch));
-		}
-
-		if (request.ifMatch != null) {
-			request.headers.put("If-Match", "\"" + entitytag(request.ifMatch));
-		}
+			if (request.ifMatch != null) {
+				request.headers.put("If-Match", "\"" + entitytag(request.ifMatch));
+			}
 
-		if (request.ifModifiedSince > 0) {
-			request.headers.put("If-Modified-Since", httpDateFormat().format(new Date(request.ifModifiedSince)));
-		}
+			if (request.ifModifiedSince > 0) {
+				request.headers.put("If-Modified-Since", httpDateFormat().format(new Date(request.ifModifiedSince)));
+			}
 
-		if (request.ifUnmodifiedSince != 0) {
-			request.headers.put("If-Unmodified-Since", httpDateFormat().format(new Date(request.ifUnmodifiedSince)));
-		}
+			if (request.ifUnmodifiedSince != 0) {
+				request.headers.put("If-Unmodified-Since",
+					httpDateFormat().format(new Date(request.ifUnmodifiedSince)));
+			}
 
-		setHeaders(request.headers, con);
+			setHeaders(request.headers, con);
 
-		configureHttpConnection(request.verb, hcon);
+			configureHttpConnection(request.verb, hcon);
 
-		final ProgressPlugin.Task task = getTask(request);
-		try {
+			final ProgressPlugin.Task task = getTask(request);
+			try {
 
-			TaggedData td = connectWithProxy(proxy,
-				() -> doConnect(request.upload, request.download, con, hcon, request, task));
-			logger.debug("result {}", td);
-			return td;
-		} catch (Throwable t) {
-			task.done("Failed " + t, t);
-			throw t;
+				TaggedData td = connectWithProxy(proxy,
+					() -> doConnect(request.upload, request.download, con, hcon, request, task));
+				logger.debug("result {}", td);
+				return td;
+			} catch (TimeoutException toe) {
+				if (trial++ < retries) {
+					logger.debug("connection timed out. url={}, timout={}, trial={}", request.url, request.timeout,
```

Comment:
Can we make the log text indicate we are retrying? Otherwise it is the same text as when we are not retrying.